action.yml

```
@@ -205,6 +252,39 @@ runs:
       path: .tmp-ansible-collection-checkout
       persist-credentials: false
       ref: ${{ inputs.git-checkout-ref }}
+      fetch-depth: >-
+        ${{
+          steps.compute-change-detection-coverage.outputs.change-detection-arg
+          && '0' || '1'
+        }}
+
+  - name: Log the next step action
+    if: >-
+      !inputs.collection-src-directory
+      && steps.compute-change-detection-coverage.outputs.change-detection-arg
+    run: >-
+      echo ▷ Create branches for change detection
+    shell: bash
+  - name: Create branches for change detection
+    if: >-
+      !inputs.collection-src-directory
+      && steps.compute-change-detection-coverage.outputs.change-detection-arg
+    run: |
+      # Create a branch for the current HEAD, which happens to be a merge commit
+      git checkout -b __action_branch_name
```

Try doing this
```suggestion
      git branch pull-request-${{ github.event.pull_request.number }} refs/remotes/pull/${{ github.event.pull_request.number }}/merge
```

---

d05eda558a030316d29d195cfaa82970102f5aae

---

This results in
```
* (HEAD detached at pull/49/merge) 9718532 Merge d05eda558a030316d29d195cfaa82970102f5aae into ce3e83585495ccb95be59a9698a342f50e0d929a
```
and makes the run fail (https://github.com/ansible-community/ansible-test-gh-action/actions/runs/3464057464/jobs/5785116552).

---

Aha! So this is what fails:
```console
[rc=1] $ git symbolic-ref --short HEAD
fatal: ref HEAD is not a symbolic ref
```

---

Okay, let's try going with
```suggestion
      git checkout -b 'pull-request-${{ github.event.pull_request.number }}'
```
for now. I think that maybe this should be improved on the ansible-test side but this'll do until then.

---

cd80f5f6421a321ef6d57fac901baa6afa00121e.